include/swift/Runtime/Mutex.h

```diff
@@ -101,57 +131,192 @@ class ConditionVariable {
 
 private:
   ConditionHandle Handle;
-};
 
-using ScopedNotifyOne = ScopedNotifyOneT<ConditionVariable>;
-using ScopedNotifyAll = ScopedNotifyAllT<ConditionVariable>;
+public:
+  /// A Mutex object that also supports ConditionVariables.
+  ///
+  /// This is NOT a recursive mutex.
+  class Mutex {
+
+    Mutex(const Mutex &) = delete;
+    Mutex &operator=(const Mutex &) = delete;
+    Mutex(Mutex &&) = delete;
+    Mutex &operator=(Mutex &&) = delete;
```

Comment:
These don't have to be private.  The `= delete` will ensure that they cannot be called irrespective of access level.

Reply:
This pattern (having them before the `public:` block with `= delete`) is repeated throughout this file. Is there any particular reason to change this, or should we just leave it be?

Reply:
Naw, leaving it as is for this change is fine.  It was more for knowledge sharing purposes.  The original reason for these being private is that before C++11, the methods would be looked up by access control and rely on the missing definition to prevent usage.  With the newer functionality, they can be public as the compiler now enforces the prevention no matter what the access level.